CMake/Version.cmake

```diff
@@ -1,11 +1,28 @@
+set(VERSION_MAJOR 0)
+set(VERSION_MINOR 0)
+set(VERSION_PATCH 0)
+set(VERSION_COMMIT 0)
+
 find_program(GIT git)
-execute_process(
-    COMMAND ${GIT} describe
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    OUTPUT_VARIABLE GIT_DESCRIBE_DIRTY
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-string(REGEX REPLACE "^v([0-9]+)\\..*" "\\1" VERSION_MAJOR "${GIT_DESCRIBE_DIRTY}")
-string(REGEX REPLACE "^v[0-9]+\\.([0-9]+).*" "\\1" VERSION_MINOR "${GIT_DESCRIBE_DIRTY}")
-string(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" VERSION_PATCH "${GIT_DESCRIBE_DIRTY}")
-string(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.[0-9]+\\-([0-9]+).*" "\\1" VERSION_COMMIT "${GIT_DESCRIBE_DIRTY}")
+if(GIT)
```

Comment:
Isn't the problem that there's no `.git` dir, not that the `git` command is not present? Though it would be good to not run this code if either one is not present.

Reply:
The original problem is caused by the absence of `.git` directory, indeed. But the user story is that source code is acquired by the `wget` command, not `git clone`. So, I assumed that `git` command may not be present on user's machine too.

Reply:
Yes, neither one may be present. So both should be tested for?

Reply:
That's what this patch does.

Reply:
Ah I was expecting a `find_file` or similar, but I guess `git describe` will just fail.

Reply:
Yes, that was the original issue regarding #5874.